packages/mui-system/src/breakpoints/breakpoints.js

```diff
@@ -121,10 +121,19 @@ export function createEmptyBreakpointObject(breakpointsInput = {}) {
 
 export function removeUnusedBreakpoints(breakpointKeys, style) {
   return breakpointKeys.reduce((acc, key) => {
+    const zeroMediaQueryRgx = /^@media\s\(min-width:\s?[0](px|rem|em)\)$/;
```

Comment:
We could make this work for any arbitrary unit since the unit is configurable by consumers. Not sure it's worth it though. Something like this, but being case insensitive since consumers can input units in uppercase and AFAIK we don't convert units to lowercase:

```suggestion
    const zeroMediaQueryRgx = /min-width:\s?[0][a-z]*?\)$/;
```